### PR TITLE
AI Chat: search queries learn more link

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/38709

Figma:
image

[Storybook](https://build-artifacts.brave.com/brave-browser-build-pr/ai-chat-search-learn-more/dbe404926d4db1733edd933dac54b80440a2e9ce/storybook/index.html?path=/story/chat-chat--panel):
image

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ai_chat/resources/page/components/assistant_response/index.tsx [hunk 1]
```diff
@@ -37,7 +41,7 @@ function SearchSummary (props: { searchQueries: string[] }) {
     <div className={styles.searchSummary}>
       <Icon name="brave-icon-search-color" />
       <span>
-        {message}
+        {message} <a className={styles.searchLearnMoreLink} href='#' onClick={handleLearnMore}>{getLocale('learnMore')}</a>
```
Comment (petemill) [inline components/ai_chat/resources/page/components/assistant_response/index.tsx:44]: I think it's just about ok to concat this and not have it be part of the `'searchQueries'` string

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by petemill** [comment]
oops I merged this before your review @nullhook - can you let me know if there are any issues with it?

**Review by nullhook — COMMENTED** [review]
lgtm

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.68.70 Chromium: 126.0.6478.26 (Official Build) nightly (x86_64)
-- | --
Revision | 28f8af9c7d27ab764fc358d8e3c5735d3139d267
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Confirmed `Learn more` loaded `https://support.brave.com/hc/en-us/categories/20990938292237-Brave-Leo`


example | example | example | example | example | example | example
---------|----------|---------|----------|----------|---------|----------
Screen Shot 2024-06-03 at 11 50 41 PM | Screen Shot 2024-06-03 at 11 50 45 PM | Screen Shot 2024-06-03 at 11 50 48 PM | Screen Shot 2024-06-03 at 11 50 54 PM | Screen Shot 2024-06-03 at 11 51 00 PM | Screen Shot 2024-06-03 at 11 51 02 PM | Screen Shot 2024-06-03 at 11 51 05 PM

**Pull request merged**